platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java

```diff
@@ -199,6 +200,7 @@ public int getVersion() {
     public MerkleRoot copy() {
         throwIfImmutable();
         throwIfDestroyed();
+        setImmutable(true);
```

Comment:
I'm not understanding why this wasn't necessary before, but is necessary now. Was it a bug before? If not, what about your changes makes this a requirement?

Reply:
>Was it a bug before?

Yes. It most definitely was a bug, and we didn't have any side effects of that only because this State is used in the test code, not prod. With the necessity of creating snapshots, it revealed itself. So, I fixed it.